### PR TITLE
Implement a sixel renderer

### PR DESCRIPTION
**Pull request by prashnts**
As I mentioned in #50, here's an implementation of sixel support.

Note that I added a new option `pixel_outline` to add outline around individual pixels.

Screenshot 2023-06-09 at 23 41 24

On mac use brew to [install libsixel](https://github.com/saitoha/libsixel), and a compatible terminal is required.

### CODE REVIEW

**Comment by prashnts** [comment]
It's usable for light animations. 

https://github.com/ty-porter/RGBMatrixEmulator/assets/5663391/ad5bed68-628c-412b-b462-9f99d254d180

File: RGBMatrixEmulator/emulators/options.py [hunk 1]
```diff
@@ -59,6 +59,8 @@ def __init__(self):
             Logger.warning('"{}" pixel style option not recognized. Valid options are "square", "circle". Defaulting to "square"...'.format(config_pixel_style))
 
         self.pixel_size = emulator_config.pixel_size
+        self.pixel_outline = emulator_config.DEFAULT_CONFIG.get('pixel_outline')
+        self.pixel_outline = emulator_config.pixel_outline
```
Comment (ty-porter) [inline RGBMatrixEmulator/emulators/options.py:63]: The default here is redundant, the options class merges defaults into the user-defined config on initialization

File: RGBMatrixEmulator/adapters/sixel_adapter.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+import libsixel as sixel
+
+from PIL import Image, ImageDraw, ImageEnhance
+from typing import List
+
+from RGBMatrixEmulator.adapters.base import BaseAdapter
+from RGBMatrixEmulator.graphics.color import Color
+
+
+def enlarge_pixels(pixels: List[List[Color]], scale: int, outline: int) -> Image.Image:
```
Comment (ty-porter) [inline RGBMatrixEmulator/adapters/sixel_adapter.py:13]: I would prefer these match the other adapters -- methods on the adapter class itself, prefixed by double underscore (i.e. `def __enlarge_pixels(. . .)`)

Unless there's a good reason not to?

**Review by ty-porter — COMMENTED** [review]
This looks really good, some minor feedback

File: RGBMatrixEmulator/emulators/options.py (hunk 1 above)
Comment (prashnts) [reply RGBMatrixEmulator/emulators/options.py]: Good point, I've fixed it!

File: RGBMatrixEmulator/adapters/sixel_adapter.py (hunk 2 above)
Comment (prashnts) [reply RGBMatrixEmulator/adapters/sixel_adapter.py:13]: While I'm not strongly biased to this, I think we can move the `enlarge_pixel` function to a module. That way the same drawing function can be used by both sixel and browser adapter. Perhaps in a different PR -- and in which case I am ready to move it in the class to match the codebase style. Let me know what you prefer!

Comment (ty-porter) [reply RGBMatrixEmulator/adapters/sixel_adapter.py:13]: Ok, that makes sense. I don't have issue with it as-is then

**Review by ty-porter — APPROVED** [review]

**Pull request merged**